python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+        elif self.config.doGenerateImage:
+            return self.makeImage()
+        elif self.config.doGenerateData:
+            return self.makeData()
+        else:
+            return None
+
+    def makeImage(self):
+        """Generate a simulated ISR LSST image.
+
+        Returns
+        -------
+        exposure : `lsst.afw.image.Exposure` or `dict`
+            Simulated ISR image data.
+
+        Notes
+        -----
+        This method constructs a "raw" data image.
+        """
+        exposure = self.getExposure()
+
+        for idx, amp in enumerate(exposure.getDetector()):
+
+            # Get image bbox and data
+            imageBBox = amp.getRawDataBBox().shiftedBy(amp.getRawXYOffset())
+            ampData = exposure.image[imageBBox]
+
+            # Sky effects in e-
+            if self.config.doAddSky:
+                self.amplifierAddNoise(ampData, self.config.skyLevel, np.sqrt(self.config.skyLevel))
+
+            if self.config.doAddSource:
+                for sourceAmp, sourceFlux, sourceX, sourceY in zip(self.config.sourceAmp,
+                                                                   self.config.sourceFlux,
+                                                                   self.config.sourceX,
+                                                                   self.config.sourceY):
+                    if idx == sourceAmp:
+                        self.amplifierAddSource(ampData, sourceFlux, sourceX, sourceY)
+
+            # Post ISR effects in e-
+            if self.config.doAddFringe:
+                self.amplifierAddFringe(amp, ampData, np.array(self.config.fringeScale),
+                                        x0=np.array(self.config.fringeX0),
+                                        y0=np.array(self.config.fringeY0))
+
+            if self.config.doAddFlat:
+                if ampData.getArray().sum() == 0.0:
+                    self.amplifierAddNoise(ampData, 1.0, 0.0)
+                u0 = exposure.getDimensions().getX()
+                v0 = exposure.getDimensions().getY()
+                self.amplifierMultiplyFlat(amp, ampData, self.config.flatDrop, u0=u0, v0=v0)
+
+            # ISR effects
+            # 1. Add dark in e- (different from isrMock which does it in ADU)
+            if self.config.doAddDark:
+                self.amplifierAddNoise(ampData,
+                                       self.config.darkRate * self.config.darkTime,
+                                       np.sqrt(self.config.darkRate
+                                               * self.config.darkTime))
+
+            # 2. Gain normalize  (from e- to ADU)
+            # TODO: DM-??? gain from PTC per amplifier
+            # TODO: DM-??? gain with temperature dependence
+            if self.config.doAddGain:
+                self.addGain(ampData, self.config.gain)
+
+            # 3. Add bias frame to the image region in ADU
+            if self.config.doAddBias:
```

Comment:
This the pre-existing config name from @czwa, but I think it really is adding noise associated with a bias frame, and not a bias frame, and should be commented accordingly.  (Assuming Chris agrees).

Reply:
I am now adding the bias frame (level and scatter) at this stage when no serial overscan is applied, and doing that step in the serial overscan if is applied.